stakedb/stakedb.go

```diff
@@ -331,22 +331,20 @@ func (db *StakeDatabase) PopulateLiveTicketCache() error {
 
 	log.Info("Pre-populating live ticket cache and computing pool value...")
 
-	// Send all the live ticket requests
+	// Send all the live ticket requests.
 	type promiseGetRawTransaction struct {
 		result rpcclient.FutureGetRawTransactionResult
 		ticket chainhash.Hash
 	}
 	promisesGetRawTransaction := make([]promiseGetRawTransaction, 0, len(liveTickets))
-
-	// Send all the live ticket requests
-	for _, hash := range liveTickets {
+	for i := range liveTickets {
 		promisesGetRawTransaction = append(promisesGetRawTransaction, promiseGetRawTransaction{
-			result: db.NodeClient.GetRawTransactionAsync(&hash),
-			ticket: hash,
+			result: db.NodeClient.GetRawTransactionAsync(&liveTickets[i]),
+			ticket: liveTickets[i],
```

Comment:
With the previous loop, `hash` was a copy of a `chainhash.Hash` (not a pointer to one), and when creating the promise with `&hash`, the copy was likely persisting until the promises slice was cleaned up.  Further, another copy was being made for the `tickets` field in the promise container struct, so there were too many copies.